Mathlib/Algebra/Algebra/Subalgebra/Pi.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Andrew Yang
+-/
+import Mathlib.Algebra.Algebra.Pi
+import Mathlib.Algebra.Algebra.Subalgebra.Basic
+import Mathlib.LinearAlgebra.Pi
+
+/-!
+# Products of subalgebras
+
+In this file we define the product of subalgebras as a subalgebra of the product algebra.
+
+## Main definitions
+
+ * `Subalgebra.pi`: the product of subalgebras.
+-/
+
+open Algebra
+
+namespace Subalgebra
+variable {ι R : Type*} {S : ι → Type*} [CommSemiring R] [∀ i, Semiring (S i)] [∀ i, Algebra R (S i)]
+  {t : Set ι} {s s₁ s₂ : ∀ i, Subalgebra R (S i)} {x : ∀ i, S i}
+
+/-- The product of subalgebras as a subalgebra. -/
+@[simps coe toSubsemiring]
```

Comment:
Does this generate something in terms of `Subsemiring.pi`? Can you add the `toSubmodule` lemma?

Reply:
`Subsemiring.pi` doesn't exist!

Reply:
Then this `toSemiring` lemma is probably pretty useless.

Reply:
It does generate something which makes `(Subalgebra.pi s t).toSubsemiring` simplify further. I think it's better to have it than not